sw/device/lib/dif/dif_kmac.c

```diff
@@ -765,8 +769,37 @@ dif_result_t dif_kmac_end(const dif_kmac_t *kmac,
 
 dif_result_t dif_kmac_config_is_locked(const dif_kmac_t *kmac,
                                        bool *is_locked) {
+  if (kmac == NULL || is_locked == NULL) {
+    return kDifBadArg;
+  }
+
   uint32_t reg =
       mmio_region_read32(kmac->base_addr, KMAC_CFG_REGWEN_REG_OFFSET);
   *is_locked = bitfield_bit32_read(reg, KMAC_CFG_REGWEN_EN_BIT);
   return kDifOk;
 }
+
+dif_result_t dif_kmac_get_status(const dif_kmac_t *kmac,
+                                 dif_kmac_status_t *kmac_status) {
+  if (kmac == NULL || kmac_status == NULL) {
+    return kDifBadArg;
+  }
+
+  uint32_t reg = mmio_region_read32(kmac->base_addr, KMAC_STATUS_REG_OFFSET);
+
+  bitfield_field32_t field = {.mask = 0x07, .index = KMAC_STATUS_SHA3_IDLE_BIT};
```

Comment:
In general, common style is to inline this declaration into the call to `field32_read`.

That said, I'm a bit confused about what's going on here. Shouldn't this be `bit32_read`, seeing as they're `BIT` registers?

Reply:
I've inlined the declaration.

Although they are defined as bits by the hardware it works better as fields for the software. i.e the fifo can't be empty and full at the same time.
That said, I’ve defined enums that represent the state of more than one bit, that's why I'm reading these bits as fields.
Does that make sense?